### PR TITLE
Add accordion component

### PR DESCRIPTION
**Pull request by aliabid94**
Allows users to wrap elements in an accordion. It's been requested several times and I thought this might be useful for the more complex SD demos being created right now.

Fixes: #938 

Use:
```python
with gr.Accordion("open up"):
   # components here
```

Recording 2022-09-07 at 17 04 33

### CODE REVIEW

**Comment by abidlabs** [comment]
Yesss very cool @aliabid94!

File: gradio/layouts.py [hunk 1]
```diff
@@ -296,3 +296,42 @@ def style(
...
+    Example:
+        with gradio.Accordion("See Details"):
+            gr.Markdown("lorem ipsum")
+    """
+
+    def __init__(self, label, *, open: bool = True, **kwargs):
+        """
+        Parameters:
+            label: name of accordion section.
+            open: An optional string that is assigned as the id of this component in the HTML DOM. Can be used for targeting CSS styles.
```
Comment (abidlabs) [inline gradio/layouts.py:314]: * Fix docstring for `open`

Comment (abidlabs) [reply gradio/layouts.py:314]: `open` is a Python keyword so syntax highlighters annoyingly highlight it. WDYT about naming this `closed` instead?

**Comment by abidlabs** [comment]
Would it be possible to support `gr.Accordion()` in `gr.Interface()`? The reason I ask is because a lot of `gr.Interface()` users would probably love to have a section of collapsible inputs as well. I can imagine something like the following API:

```py
gr.Interface(
  image_generator,
  [gr.Textbox(),
   gr.Accordion("Advanced Options", [
     gr.Slider(),
     gr.Slider(),
    ]
  ]
  gr.Image()
]
```

Otherwise, LGTM @aliabid94!

**Comment by abidlabs** [comment]
One other thing: the label of the Accordion is much bigger, darker, and bolder than the label of the other components as you can see here:

image

I think it stands out *too* much. What about making it the same font style as the title of tabs:

image

**Comment by aliabid94** [comment]
ok made the label smaller and not bold. Also added mention to guides. Adding to Interface will be a lot of work, let's make a separate issue for that. Let's get this in for 3.3 @abidlabs

File: guides/3)building_with_blocks/2)controlling_layout.md [hunk 2]
```diff
@@ -31,7 +31,7 @@ $demo_rows_and_columns
 
 See how the first column has two Textboxes arranged vertically. The second column has an Image and Button arranged vertically. Notice how the relative widths of the two columns is set by the `scale` parameter. The column with twice the `scale` value takes up twice the width.
 
-## Tabs
+## Tabs amd Accordions
```
Comment (abidlabs) [inline guides/3)building_with_blocks/2)controlling_layout.md:34]:
```suggestion
## Tabs and Accordions
```

File: guides/3)building_with_blocks/2)controlling_layout.md [hunk 3]
```diff
@@ -40,6 +40,8 @@ For example:
 $code_blocks_flipper
 $demo_blocks_flipper
 
+Also note the Accordion that can be opened and closed - another layout element to selectively show content.
```
Comment (abidlabs) [inline guides/3)building_with_blocks/2)controlling_layout.md:43]:
```suggestion
Also note the `gradio.Accordion('label')` in this example. The Accordion is a layout that can be toggled open or close. Like `Tabs`, it is another layout element that can selectively hide or show content. Any components that are defined inside of a `with gradio.Accordion('label'):` will be hidden or shown when the accordion's toggle icon is clicked.
```

**Comment by abidlabs** [comment]
Ok sounds good! LGTM

**Review by abidlabs — APPROVED** [review]

**Comment by aliabid94** [comment]
oh whoops merged before reading comments, will make another pr for them

**Comment by pngwn** [comment]
I missed this PR but I don't understand why this component was added instead of just adding some options to `gr.Group`, those were the conversations we had in the past and wouldn't have significantly increased API surface area like a new component does.

**Comment by abidlabs** [comment]
Agree with communicating on this more, but from my understanding `gr.Group()` is designed to change the padding/display of children. `gr.Accordion()` has a pretty orthogonal functionality (toggling the children), so I think it's clearer to have a separate class for this.

**Comment by pngwn** [comment]
I may have meant `gr.Box` which is mostly visual and could easily have taken some additional arguments.

There are also some issues with the implementation, I'll take a look at them when I get time.

**Comment by aliabid94** [comment]
Oh sorry yeah. Makes sense to incorporate into `gr.Box`

**Pull request merged**